### PR TITLE
Add support for generic column-level metadata

### PR DESCRIPTION
**Pull request by jwildfire**
Adding support in both codebook and explorer.  #149 and #169.

### CODE REVIEW

**Comment by jwildfire** [comment]
Ready for your review, @samussiah.

File: src/codebook/summaryTable/renderRow/details/renderValues.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      return i >= nValues - 3;
+    });
+    var valList = d3merge([minValues, ['...'], maxValues]);
+
+    var valueItems = list.selectAll('li').data(valList).enter().append('li');
+
+    valueItems.append('div').attr('class', 'label').text(function(d, i) {
+      return i == 0 ? 'Min' : i == valList.length - 1 ? 'Max' : ' ';
+    });
+    valueItems
```
Comment (samussiah) [inline src/codebook/summaryTable/renderRow/details/renderValues.js]: I'd right-align the text underneath Min.

File: src/codebook/summaryTable/renderRow/details/renderValues.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      }); // sort low to high
+
+    var minValues = sortedValues.filter(function(d, i) {
+      return i < 3;
+    });
+    var nValues = sortedValues.length;
+    var maxValues = sortedValues.filter(function(d, i) {
+      return i >= nValues - 3;
+    });
+    var valList = d3merge([minValues, ['...'], maxValues]);
```
Comment (samussiah) [inline src/codebook/summaryTable/renderRow/details/renderValues.js]: Edge case: don't include `'...'` if `nValues <= 6`.

Comment (samussiah) [inline src/codebook/summaryTable/renderRow/details/renderValues.js]: image

File: src/codebook/util/setDefaults.js [hunk 3]
```diff
@@ -43,6 +59,25 @@ export function setDefaults(codebook) {
...
+    //Keep the default label if the user hasn't specified a label for the column
+    defaultSettings.variableLabels.forEach(function(defaultLabel) {
+      if (userLabelVars.indexOf(defaultLabel.value_col) == -1) {
+        codebook.config.variableLabels.push(defaultLabel);
+      }
+    });
+  } else {
+    //otherwise just use whichever is populated or fallback to an empty object
+    codebook.config.variableLabels =
+      codebook.config.variableLabels || defaultSettings.variableLabels;
```
Comment (samussiah) [inline src/codebook/util/setDefaults.js]: `codebook.config.variableLabels` returns true even if it's empty.

File: src/codebook/summaryTable/renderRow/details/renderValues.js (hunk 2 above)
Comment (jwildfire) [inline src/codebook/summaryTable/renderRow/details/renderValues.js]: Done in 5a37f79bee8a06f193046fcf8568b7deae129819

File: src/codebook/util/setDefaults.js (hunk 3 above)
Comment (jwildfire) [inline src/codebook/util/setDefaults.js]: hmm, so `[]||"test" == []", but `undefined||"test" == "test"` - didn't realize that. I guess we need to just write this out to be more verbose.

File: src/codebook/summaryTable/renderRow/details/renderValues.js (hunk 1 above)
Comment (jwildfire) [inline src/codebook/summaryTable/renderRow/details/renderValues.js]: used a min-width instead: afd718363c0bc46e237876b3f0b73a485fad48fe

File: src/codebook/util/setDefaults.js (hunk 3 above)
Comment (jwildfire) [inline src/codebook/util/setDefaults.js]: Like so? c6890ca49a277bb399ec1b159ad960d9fc568151

**Comment by jwildfire** [comment]
@samussiah ready for a second look. Dealt with merge in b009ba9efc10148b13d4b5a848f211a1142d3f02 so go ahead and merge to dev if updates resolve the issues you found.

**Pull request merged**